<!DOCTYPE html>
<html>
	<head>
		<meta charset="utf-8">
		
		<title>CVE-2020-26250 - Vulnerability Data</title>
		
		<link rel="shortcut icon" type="image/png" href="/favicon.ico" />
		<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css">
		  <link href="https://fonts.googleapis.com/icon?family=Material+Icons" rel="stylesheet">
		<script src="https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/js/materialize.min.js"></script>
		<!-- Global site tag (gtag.js) - Google Analytics -->
<script async src="https://www.googletagmanager.com/gtag/js?id=G-XL44S2GJJ7"></script>
<script>
  window.dataLayer = window.dataLayer || [];
  function gtag(){dataLayer.push(arguments);}
  gtag('js', new Date());

  gtag('config', 'G-XL44S2GJJ7');
</script>
	</head>
<body>
	<div class="container">
	

  <nav>
    <div class="nav-wrapper">
      <div class="col s12">
        <a href="https://www.vulnerabilitydata.com" class="breadcrumb">Vulnerability Data</a>
        <a href="/2020" class="breadcrumb">2020</a>
        <a href="#" class="breadcrumb">CVE-2020-26250</a>
      </div>
    </div>
  </nav>

<h2>Vulnerability Data</h2>
<h1>CVE-2020-26250</h1>
<p>OAuthenticator is an OAuth login mechanism for JupyterHub. In oauthenticator from version 0.12.0 and before 0.12.2, the deprecated (in jupyterhub 1.2) configuration `Authenticator.whitelist`, which should be transparently mapped to `Authenticator.allowed_users` with a warning, is instead ignored by OAuthenticator classes, resulting in the same behavior as if this configuration has not been set. If this is the only mechanism of authorization restriction (i.e. no group or team restrictions in configuration) then all authenticated users will be allowed. Provider-based restrictions, including deprecated values such as `GitHubOAuthenticator.org_whitelist` are **not** affected. All users of OAuthenticator 0.12.0 and 0.12.1 with JupyterHub 1.2 (JupyterHub Helm chart 0.10.0-0.10.5) who use the `admin.whitelist.users` configuration in the jupyterhub helm chart or the `c.Authenticator.whitelist` configuration directly. Users of other deprecated configuration, e.g. `c.GitHubOAuthenticator.team_whitelist` are **not** affected. If you see a log line like this and expect a specific list of allowed usernames: "[I 2020-11-27 16:51:54.528 JupyterHub app:1717] Not using allowed_users. Any authenticated user will be allowed." you are likely affected. Updating oauthenticator to 0.12.2 is recommended. A workaround is to replace the deprecated `c.Authenticator.whitelist = ...` with `c.Authenticator.allowed_users = ...`. If any users have been authorized during this time who should not have been, they must be deleted via the API or admin interface, per the referenced documentation.</p>

<dl>
	<dt>Published</dt>
	<dd>2020-12-01</dd>

	<dt>Modified</dt>
	<dd>2020-12-01</dd>
</dl>

<h3>References</h3>
<ul>
	<li><a href="https://github.com/jupyterhub/oauthenticator/security/advisories/GHSA-384w-5v3f-q499">CONFIRM:https://github.com/jupyterhub/oauthenticator/security/advisories/GHSA-384w-5v3f-q499</a></li>
<li><a href="https://github.com/jupyterhub/oauthenticator/blob/master/docs/source/changelog.md#0122---2020-11-30">MISC:https://github.com/jupyterhub/oauthenticator/blob/master/docs/source/changelog.md#0122---2020-11-30</a></li>
<li><a href="https://github.com/jupyterhub/oauthenticator/commit/a4aac191c16cf6281f3d346615aefa75702b02d7">MISC:https://github.com/jupyterhub/oauthenticator/commit/a4aac191c16cf6281f3d346615aefa75702b02d7</a></li>
<li><a href="https://jupyterhub.readthedocs.io/en/1.2.2/getting-started/authenticators-users-basics.html#add-or-remove-users-from-the-hub">MISC:https://jupyterhub.readthedocs.io/en/1.2.2/getting-started/authenticators-users-basics.html#add-or-remove-users-from-the-hub</a></li>
</ul>

<footer>
	<p><a href="https://reconmap.org">Track security vulnerabilities</a> with <strong>Reconmap</strong>.</p>
</footer>


	</div>
</body>
</html>